pkg/aggregator/aggregator.go

```diff
@@ -239,6 +246,13 @@ func MergeSpecs(dest, source *spec.Swagger) error {
 
 func mergeSpecs(dest, source *spec.Swagger, renameModelConflicts, ignorePathConflicts bool) (err error) {
 	specCloned := false
+	// Paths may be empty, due to [ACL constraints](http://goo.gl/8us55a#securityFiltering).
+	if source.Paths == nil {
```

Comment:
Now that I am thinking more about this, if the source path is empty, we should not do anything. It is true that in current implementation, we do merge definitions (which we should fix) but when a source spec does not have any path, that means none of the definitions are used thus we should not do anything. just return.